packaging/os/pulp_repo.py

```diff
@@ -0,0 +1,556 @@
+
+    if state == 'publish':
+        if not repo_exists:
+            module.fail_json(msg="Repository was not found. The repository can not be published.")
+
+        if not module.check_mode:
+            server.publish_repo(repo, publish_distributor)
+
+        changed = True
+
+    if state == 'present':
+        if not repo_exists:
+            if not module.check_mode:
+                server.create_repo(
+                    repo_id=repo,
+                    relative_url=relative_url,
+                    feed_url=feed_url,
+                    serve_http=serve_http,
+                    serve_https=serve_https,
+                    proxy_host=proxy_host,
+                    proxy_port=proxy_port,
+                    ssl_ca_cert=importer_ssl_ca_cert,
+                    ssl_client_cert=importer_ssl_client_cert,
+                    ssl_client_key=importer_ssl_client_key)
+
+            changed = True
+
+        else:
+            # Check to make sure all the settings are correct
+            # The importer config gets overwritten on set and not updated, so we set the whole config at the same time.
+            if not server.compare_repo_importer_config(
+                repo,
+                feed=feed_url,
+                proxy_host=proxy_host,
+                proxy_port=proxy_port,
+                ssl_ca_cert=importer_ssl_ca_cert,
+                ssl_client_cert=importer_ssl_client_cert,
+                ssl_client_key=importer_ssl_client_key
+            ):
+                if not module.check_mode:
+                    server.update_repo_importer_config(
+                        repo,
+                        feed=feed_url,
+                        proxy_host=proxy_host,
+                        proxy_port=proxy_port,
+                        ssl_ca_cert=importer_ssl_ca_cert,
+                        ssl_client_cert=importer_ssl_client_cert,
+                        ssl_client_key=importer_ssl_client_key)
+
+                changed = True
+
+            if relative_url is not None:
+                if not server.compare_repo_distributor_config(repo, relative_url=relative_url):
+                    if not module.check_mode:
+                        server.update_repo_distributor_config(repo, relative_url=relative_url)
+
+                    changed = True
+
+            if not server.compare_repo_distributor_config(repo, http=serve_http):
+                if not module.check_mode:
+                    server.update_repo_distributor_config(repo, http=serve_http)
+
+                changed = True
+
+            if not server.compare_repo_distributor_config(repo, https=serve_https):
+                if not module.check_mode:
+                    server.update_repo_distributor_config(repo, https=serve_https)
+
+                changed = True
+
+    module.exit_json(changed=changed, repo=repo)
+
+import json
+
+# import module snippets
+from ansible.module_utils.basic import *
+from ansible.module_utils.urls import *
```

Comment:
Please move the import to the top of the file. Here are more details: https://github.com/ansible/ansible-modules-extras/pull/2369#issuecomment-223579676